drivers/wireless/bluetooth/bt_rpmsghci_server.c

```diff
@@ -0,0 +1,547 @@
+                                  FAR void *data, size_t len,
+                                  uint32_t src, FAR void *priv);
+static int rpmsghci_send_handler(FAR struct rpmsg_endpoint *ept,
+                                 FAR void *data, size_t len,
+                                 uint32_t src, FAR void *priv);
+static int rpmsghci_ioctl_handler(FAR struct rpmsg_endpoint *ept,
+                                  FAR void *data, size_t len,
+                                  uint32_t src, FAR void *priv);
+static int rpmsghci_default_handler(FAR struct rpmsg_endpoint *ept,
+                                    FAR void *data, size_t len,
+                                    uint32_t src, FAR void *priv);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const rpmsg_ept_cb g_rpmsghci_handler[] =
+{
+  rpmsghci_open_handler,       /* RPMSGHCI_OPEN */
+  rpmsghci_close_handler,      /* RPMSGHCI_CLOSE */
+  rpmsghci_send_handler,       /* RPMSGHCI_SEND */
+  rpmsghci_ioctl_handler,      /* RPMSGHCI_IOCTL */
+  rpmsghci_default_handler,    /* RPMSGHCI_RECV */
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rpmsghci_get_tx_payload_buffer
+ ****************************************************************************/
+
+static FAR void *
+rpmsghci_get_tx_payload_buffer(FAR struct rpmsghci_server_s *priv,
+                               FAR uint32_t *len)
+{
+  return rpmsg_get_tx_payload_buffer(&priv->ept, len, true);
+}
+
+/****************************************************************************
+ * Name: rpmsghci_send
+ *
+ * Description:
+ *   Send the rpmsg data and wait for ACK.
+ *
+ * Parameters:
+ *   priv    - rpmsg device handle
+ *   command - the command, RPMSGHCI_OPEN, RPMSGHCI_CLOSE, RPMSGHCI_SEND,
+ *                          RPMSGHCI_IOCTL
+ *   msg     - the message header
+ *   len     - length of the payload
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int rpmsghci_send(FAR struct rpmsghci_server_s *priv,
+                         uint32_t command, FAR struct rpmsghci_header_s *msg,
+                         int len)
+{
+  struct rpmsghci_cookie_s cookie;
+  int                      ret = OK;
+
+  memset(&cookie, 0, sizeof(struct rpmsghci_cookie_s));
+  nxsem_init(&cookie.sem, 0, 0);
+
+  msg->command = command;
+  msg->result  = -ENXIO;
+  msg->cookie  = (uintptr_t)&cookie;
+
+  ret = rpmsg_send_nocopy(&priv->ept, msg, len);
+  if (ret < 0)
+    {
+      goto errout;
+    }
+
+  ret = rpmsg_wait(&priv->ept, &cookie.sem);
```

Comment:
the response shouldn't wait the response again.

Reply:
`rpmsghci_send()` is used to send async events from HCI controller (`rpmsghci_bt_receive()`,  which is called by `bt_netdev_receive()` macro from low-level HCI controller implementation). Responses are handled with `rpmsg_send()`